pkg/helm/watcher/controller/helm_watcher.go

```diff
@@ -138,3 +158,84 @@ func (r *HelmWatcherReconciler) reconcileDelete(ctx context.Context, repository
 	// Stop reconciliation as the object is being deleted
 	return ctrl.Result{}, nil
 }
+
+// event emits a Kubernetes event and forwards the event to notification controller if configured.
+func (r *HelmWatcherReconciler) event(ctx context.Context, hr *sourcev1.HelmRepository, revision, severity, msg string) {
```

Comment:
```suggestion
// sendEvent sends a Kubernetes event and forwards the event to notification controller if configured.
func (r *HelmWatcherReconciler) sendEvent(ctx context.Context, hr *sourcev1.HelmRepository, revision, severity, msg string) {
```

Reply:
This is a really good question, and actually, a tricker one to answer than it seems...

I'd say "only installed ones", but determining that is harder than it looks, let's get it going for all, and then work out to filter for "only installed ones".